ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/Conversions.scala

```diff
@@ -241,6 +241,16 @@ object Conversions {
   def parseHash(bytes: com.google.protobuf.ByteString): crypto.Hash =
     crypto.Hash.assertFromBytes(data.Bytes.fromByteString(bytes))
 
+  def extractHashFromArchive(rawArchive: Raw.Archive): String =
+    com.daml.lf.archive.ArchiveParser.assertFromByteString(rawArchive.bytes).getHash
```

Comment:
Given that the hash is a small part of the archive, I suggest a custom parser here (or in `ArchiveParser`).

Does `ArchiveParser` actually exist?

Reply:
This is actually the approach we discussed with @remyhaemmerle-da. WDYT? Also, this will go to the `kv-support` soon, I believe.

Reply:
@remyhaemmerle-da We can make a custom parser based on the following protobuf. Whether it’s significantly faster I dunno but at least it’s plausible.
```
message Archive {
  string hash = 4;
}
```

Reply:
please put optimization in kv-support. the language-team may move those latter in the archive module.

Reply:
> @remyhaemmerle-da We can make a custom parser based on the following protobuf. Whether it’s significantly faster I dunno but at least it’s plausible.
I am completely fine for any optimization, as long one have sufficient confidence it is actually faster. 

If we are so concern about performance, I will also skip the string conversion. 
```
message Archive {
  bytes hash = 4;
}
```

Reply:
> I would expect that decoding a field to a `ByteString` is pretty cheap so I’m not sure the performance impact of this is particularly large.

Depends how you got there. `CodedInputStream#readBytes` might just take a substring, or it might copy, depending on the underlying data structure.

Reply:
Here the underlying is a `ByteString`, isn't it ? 
So I expect it take a substring. 
If you can show me that in you usecase, it is copying I am fine for writing an optimization.

Reply:
I am good with holding off on this until we actually need it (if ever).